DllImportGenerator/DllImportGenerator/Marshalling/MarshallingGenerator.cs

```diff
@@ -303,7 +305,7 @@ private static IMarshallingGenerator CreateStringMarshaller(TypePositionInfo inf
             throw new MarshallingNotSupportedException(info, context);
         }
         
-        private static ExpressionSyntax GetNumElementsExpressionFromMarshallingInfo(TypePositionInfo info, StubCodeContext context)
+        private static ExpressionSyntax GetNumElementsExpressionFromMarshallingInfo(TypePositionInfo info, StubCodeContext context, AnalyzerConfigOptions options)
```

Comment:
Help me defend not exposing the `AnalyzerConfigOptions` on `StubCodeContext`. I am mostly there but want to hear another argument.

Reply:
Here's an argument for not putting it on the context:

The options are part of the configuration of the generators. Technically, the context is not only part of the configuration of the generators, but the execution as well. Since the options don't change over time but the context object might, we should initialize the marshallers with the options at selection time instead of at generation time.

Additionally, in the future the `UseMarshalType` option will become a no-op when we ship, so we won't need to pass down the options to the specific generators. At which point, it wouldn't make sense for the context to carry the options with it.

Reply:
Done.